app/controllers/recipes_controller.rb

```diff
@@ -6,7 +6,7 @@ def index
   end
 
   def show
-    if current_user != Recipe.find(params[:id]).user
+    if current_user != Recipe.find(params[:id]).user && !Recipe.find(params[:id]).public
       flash[:notice] = 'You are not authorized to access this page'
       redirect_to recipes_path
     end
```

Comment:
## Change Required
This doesn't work. I cannot see the page if I am not logged in.